libs/libcommon/include/common/DateLUT.h

```diff
@@ -10,9 +10,7 @@
 #include <mutex>
 #include <memory>
 
-/** Этот класс предоставляет метод для того, чтобы создать объект DateLUTImpl
-  * для заданного часового пояса, если он не существует.
-  */
+/// This class provides lazy initialization and lookup of singleton DateLUTImpl objects for a given timezone.
 class DateLUT : public Singleton<DateLUT>
```

Comment:
Some performance measurements similar to those done in METR-15618 (taken in the VirtualBox VM on my laptop so take with a grain of salt). Measurements are: old throughput -> new throughput.

DateTime serialization with the default timezone:
`SELECT count() FROM system.numbers WHERE NOT ignore(toString(now() + number % 500000000))`
185 MB/s. -> 185 MB/s. (no noticeable difference)

DateTime serialization with conversion into arbitrary timezone:
`SELECT count() FROM system.numbers WHERE NOT ignore(toString(now() + number % 500000000, 'Europe/London'))`
134 MB/s. -> 135 MB/s. (no noticeable difference)

DateTime serialization with conversion into the default timezone:
`SELECT count() FROM system.numbers WHERE NOT ignore(toString(now() + number % 500000000, 'Europe/Moscow'))`
186 MB/s. -> 172 MB/s. (~8% slower)
`SELECT count() FROM system.numbers WHERE NOT ignore(toString(now() + number % 500000000, ''))`
186 MB/s. -> 172 MB/s. (the same, ~8% slower)

Note that measurements with timezone conversions in METR-15618 are bogus due to incorrect queries.

Is the slowdown ok considering that the common case is ok? I've tracked it down to DB/Functions/FunctionsConversion.h:convertTimestamp() function which is absent from the profile of original server performing the same queries. Apparently, the compiler isn't able to properly optimize equality comparison in this function because one or both of the values is loaded from an atomic.